In [2]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

df = pd.read_csv("../data/raw/fraud_oracle.csv")   # replace with your real filename
print(df.shape)
df.head()

FileNotFoundError: [Errno 2] No such file or directory: '../data/raw/fraud_oracle.csv'

In [3]:
import os
print(os.listdir("../data/raw"))

FileNotFoundError: [WinError 3] The system cannot find the path specified: '../data/raw'

In [4]:
import os
print(os.getcwd())

C:\Users\N6N0C\Documents\insurance-fraud-detection


In [1]:
import os
print(os.listdir("../data/raw"))

['fraud_oracle.csv']


In [2]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

df = pd.read_csv("../data/raw/fraud_oracle.csv")   # adjust if the exact name differs
print(df.shape)
df.head()

(15420, 33)


,Month,WeekOfMonth,DayOfWeek,Make,AccidentArea,DayOfWeekClaimed,MonthClaimed,WeekOfMonthClaimed,Sex,MaritalStatus,Age,Fault,PolicyType,VehicleCategory,VehiclePrice,FraudFound_P,PolicyNumber,RepNumber,Deductible,DriverRating,Days_Policy_Accident,Days_Policy_Claim,PastNumberOfClaims,AgeOfVehicle,AgeOfPolicyHolder,PoliceReportFiled,WitnessPresent,AgentType,NumberOfSuppliments,AddressChange_Claim,NumberOfCars,Year,BasePolicy
0,Dec,5,Wednesday,Honda,Urban,Tuesday,Jan,1,Female,Single,21,Policy Holder,Sport - Liability,Sport,more than 69000,0,1,12,300,1,more than 30,more than 30,none,3 years,26 to 30,No,No,External,none,1 year,3 to 4,1994,Liability
1,Jan,3,Wednesday,Honda,Urban,Monday,Jan,4,Male,Single,34,Policy Holder,Sport - Collision,Sport,more than 69000,0,2,15,400,4,more than 30,more than 30,none,6 years,31 to 35,Yes,No,External,none,no change,1 vehicle,1994,Collision
2,Oct,5,Friday,Honda,Urban,Thursday,Nov,2,Male,Married,47,Policy Holder,Sport - Collision,Sport,more than 69000,0,3,7,400,3,more than 30,more than 30,1,7 years,41 to 50,No,No,External,none,no change,1 vehicle,1994,Collision
3,Jun,2,Saturday,Toyota,Rural,Friday,Jul,1,Male,Married,65,Third Party,Sedan - Liability,Sport,20000 to 29000,0,4,4,400,2,more than 30,more than 30,1,more than 7,51 to 65,Yes,No,External,more than 5,no change,1 vehicle,1994,Liability
4,Jan,5,Monday,Honda,Urban,Tuesday,Feb,2,Female,Single,27,Third Party,Sport - Collision,Sport,more than 69000,0,5,3,400,1,more than 30,more than 30,none,5 years,31 to 35,No,No,External,none,no change,1 vehicle,1994,Collision


In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 15420 entries, 0 to 15419
Data columns (total 33 columns):
 #   Column                Non-Null Count  Dtype
---  ------                --------------  -----
 0   Month                 15420 non-null  str  
 1   WeekOfMonth           15420 non-null  int64
 2   DayOfWeek             15420 non-null  str  
 3   Make                  15420 non-null  str  
 4   AccidentArea          15420 non-null  str  
 5   DayOfWeekClaimed      15420 non-null  str  
 6   MonthClaimed          15420 non-null  str  
 7   WeekOfMonthClaimed    15420 non-null  int64
 8   Sex                   15420 non-null  str  
 9   MaritalStatus         15420 non-null  str  
 10  Age                   15420 non-null  int64
 11  Fault                 15420 non-null  str  
 12  PolicyType            15420 non-null  str  
 13  VehicleCategory       15420 non-null  str  
 14  VehiclePrice          15420 non-null  str  
 15  FraudFound_P          15420 non-null  int64
 16  PolicyNumber   

In [4]:
df.isna().sum().sort_values(ascending=False)

Month                   0
WeekOfMonth             0
DayOfWeek               0
Make                    0
AccidentArea            0
DayOfWeekClaimed        0
MonthClaimed            0
WeekOfMonthClaimed      0
Sex                     0
MaritalStatus           0
Age                     0
Fault                   0
PolicyType              0
VehicleCategory         0
VehiclePrice            0
FraudFound_P            0
PolicyNumber            0
RepNumber               0
Deductible              0
DriverRating            0
Days_Policy_Accident    0
Days_Policy_Claim       0
PastNumberOfClaims      0
AgeOfVehicle            0
AgeOfPolicyHolder       0
PoliceReportFiled       0
WitnessPresent          0
AgentType               0
NumberOfSuppliments     0
AddressChange_Claim     0
NumberOfCars            0
Year                    0
BasePolicy              0
dtype: int64

In [5]:
print(df["FraudFound_P"].value_counts())
print(df["FraudFound_P"].value_counts(normalize=True).round(3))

FraudFound_P
0    14497
1      923
Name: count, dtype: int64
FraudFound_P
0    0.94
1    0.06
Name: proportion, dtype: float64


In [6]:
print("Duplicate rows:", df.duplicated().sum())
print("Columns:", df.columns.tolist())

Duplicate rows: 0
Columns: ['Month', 'WeekOfMonth', 'DayOfWeek', 'Make', 'AccidentArea', 'DayOfWeekClaimed', 'MonthClaimed', 'WeekOfMonthClaimed', 'Sex', 'MaritalStatus', 'Age', 'Fault', 'PolicyType', 'VehicleCategory', 'VehiclePrice', 'FraudFound_P', 'PolicyNumber', 'RepNumber', 'Deductible', 'DriverRating', 'Days_Policy_Accident', 'Days_Policy_Claim', 'PastNumberOfClaims', 'AgeOfVehicle', 'AgeOfPolicyHolder', 'PoliceReportFiled', 'WitnessPresent', 'AgentType', 'NumberOfSuppliments', 'AddressChange_Claim', 'NumberOfCars', 'Year', 'BasePolicy']


In [7]:
print(df["PolicyNumber"].nunique(), "unique values out of", len(df), "rows")
print(df["RepNumber"].value_counts().head(10))
print(df["Year"].value_counts())
print(df["Days_Policy_Accident"].value_counts())
print(df["PastNumberOfClaims"].value_counts())
print(df["AgeOfVehicle"].value_counts())
print(df["PoliceReportFiled"].value_counts())
print(df["WitnessPresent"].value_counts())

15420 unique values out of 15420 rows
RepNumber
7     1069
9      999
1      987
5      987
10     986
12     977
15     977
16     967
2      956
3      949
Name: count, dtype: int64
Year
1994    6142
1995    5195
1996    4083
Name: count, dtype: int64
Days_Policy_Accident
more than 30    15247
none               55
8 to 15            55
15 to 30           49
1 to 7             14
Name: count, dtype: int64
PastNumberOfClaims
2 to 4         5485
none           4352
1              3573
more than 4    2010
Name: count, dtype: int64
AgeOfVehicle
7 years        5807
more than 7    3981
6 years        3448
5 years        1357
new             373
4 years         229
3 years         152
2 years          73
Name: count, dtype: int64
PoliceReportFiled
No     14992
Yes      428
Name: count, dtype: int64
WitnessPresent
No     15333
Yes       87
Name: count, dtype: int64


## Data Overview
- 15,420 rows, 33 columns, 0 missing values, 0 duplicates — a clean dataset
- Target: FraudFound_P (0 = no fraud, 1 = fraud)
- Fraud rate: 6.0% (923 of 15,420 claims) — imbalanced; precision/recall/PR-AUC prioritized over accuracy

### Column notes
- PolicyNumber: 15,420/15,420 unique — pure ID, excluded from modeling
- RepNumber: 16 categories, evenly distributed — kept as categorical, weak predictor expected
- Year: only 1994-1996 — old dataset; noted as a limitation
- Days_Policy_Accident: 98.9% "more than 30" — very low variance, likely weak predictor
- PastNumberOfClaims, AgeOfVehicle: ordinal categories, need custom sort order for charts
- PoliceReportFiled: 2.8% "Yes" — tests H4, but rare-category imbalance itself
- WitnessPresent: 0.6% "Yes" — extremely rare, limited statistical power expected

### Known Limitations
- Data is from 1994-1996 — a synthetic/older dataset, not representative of current fraud patterns
- FraudFound_P imbalance (6%) requires careful metric choice (precision, recall, PR-AUC, not accuracy)
- Findings show association, not proof that a factor causes fraud